### PR TITLE
Flag for saving environent

### PR DESCRIPTION
**Pull request by hieuvubk**
Closes: #4654 

## What is the purpose of the change

Refer from [cmt](https://github.com/osmosis-labs/osmosis/issues/4654#issue-1629502654).

One limitation of using environment variable is that we have to save to `.omsosisd`. This can cause trouble for users who don't want to use this feature.

So I added a bool flag `init-env` to `InitCmd` to determine if the user wants to use the current directory as environment variable. If not, `home` flag is used normally and no `.env` file is created. Then it is possible for users to set environment variables with the `set-env` command

## Brief Changelog

*(for example:)*
 
  - *The metadata is stored in the blob store on job creation time as a persistent artifact*
  - *Deployments RPC transmits only the blob storage reference*
  - *Daemons retrieve the RPC data from the blob cache*

## Testing and Verifying

*(Please pick one of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added unit test that validates ...*
  - *Added integration tests for end-to-end deployment with ...*
  - *Extended integration test for ...*
  - *Manually verified the change by ...*

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: cmd/osmosisd/cmd/init.go [hunk 1]
```diff
@@ -36,6 +36,9 @@ const (
 
 	// FlagSeed defines a flag to initialize the private validator key from a specific seed.
 	FlagRecover = "recover"
+
+	// FlagInitEnv defines a flag to create environment file & save current home directory into it.
+	FlagInitEnv = "init-env"
```
Comment (mattverse) [inline cmd/osmosisd/cmd/init.go:41]:
```suggestion
	FlagSetEnv = "set-env"
```

File: cmd/osmosisd/cmd/init.go [hunk 2]
```diff
@@ -181,6 +187,7 @@ func InitCmd(mbm module.BasicManager, defaultNodeHome string) *cobra.Command {
 	cmd.Flags().BoolP(FlagOverwrite, "o", false, "overwrite the genesis.json file")
 	cmd.Flags().Bool(FlagRecover, false, "provide seed phrase to recover existing key instead of creating")
 	cmd.Flags().String(flags.FlagChainID, "", "genesis file chain-id, if left blank will be randomly created")
+	cmd.Flags().Bool(FlagInitEnv, false, "create environment file & save current home directory into it")
```
Comment (mattverse) [inline cmd/osmosisd/cmd/init.go:190]:
```suggestion
	cmd.Flags().Bool(FlagSetEnv, false, "set and save current directory as home directory")
```

**Review by mattverse — COMMENTED** [review]
tACK. Confirmed that this is working as expected via using binary built from current branch.

Leaving one suggestion for the flag name, since the flag name had me confused for quite a while on what this feature does. Lmk if you think the suggested change would make sense!

**Review by mattverse — APPROVED** [review]
Nice Work! LGTM

**Review by p0mvn — APPROVED** [review]
Thanks

**Pull request merged**